src/govuk/common/index.mjs

```diff
@@ -61,16 +61,21 @@ export function mergeConfigs (/* configObject1, configObject2, ...configObjects
    * each of our objects, nor transform our dataset from a flat list into a
    * nested object.
    *
-   * @param {object} configObject - Deeply nested object
-   * @returns {object} Flattened object with dot-separated keys
+   * @param {Object<string, any>} configObject - Deeply nested object
+   * @returns {Object<string, string>} Flattened object with dot-separated keys
```

Comment:
Great spot, and it's found another potentially crashing bug too

We're trusting that translation "strings" in `I18n()` are always strings
https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/i18n.mjs#L46

I might try and use `unknown` (checks needed) instead of `any` (doesn't matter)

Reply:
Yep all works with `Object<string, unknown>` 👍 

I've removed the `TranslationsFlattened` type as it was fibbing 😬 

```mjs
/**
 * Translated messages (flattened)
 *
 * @private
 * @typedef {Object<string, string> | {}} TranslationsFlattened
 */
```

You'll see some extra commits in the other PR now to check for string/number/boolean values:

* [`c19ed21` **Filters out non-string "Placeholder values"**](https://github.com/alphagov/govuk-frontend/pull/2987/commits/c19ed21460bf1c759ba994f985e141501c41542c)
* [`57b465d` **Filters out non-string "Translation strings"**](https://github.com/alphagov/govuk-frontend/pull/2987/commits/57b465d0dc4d84f04da94126249e3fc0a4506ec8)